### ISSUE TITLE
Search inconsistent on Assignee field

### ISSUE THREAD

**Issue by HymaKanchiAU** [aehrc/snomio#283]
Navigate to https://dev-snomio.ihtsdotools.org/dashboard/tickets/backlog
In the Search field type Assignee: hyma kanchi
Actual: No tickets are displayed
![image](https://github.com/aehrc/snomio/assets/126848487/0282ce57-b7b5-4a94-9971-df50d7528c3d)

Expected: Tickets that have assignee as Hyma kanchi should be displayed.

Noticed that 

- Filter uses Hyma Kanchi where are Search uses hkanchi - inconsistency ( search on SI username i.e. hkanchi and not hyma kanchi)
![image](https://github.com/aehrc/snomio/assets/126848487/d2f651cb-6f85-4700-9a18-bb04b1a6eb3a)

- When you type Assignee:h for partial search displays the tickets with Assignee's who have h in them example (Hyma Kanchi, Ada Hon, Dion Athans. Is this search field using contains? 
![image](https://github.com/aehrc/snomio/assets/126848487/1c83457b-a7a7-4672-9463-b0cd7367fd3c)

Please check with @kylynnloi or @ChrisKellaleaMaynard before implementing the bug fixes.

**Comment by clinnygee** [comment]
@ChrisKellaleaMaynard 

I am not sure what to do with this one. 

I think the filter should be ignored for the time being - I think those changes should be part of a rework of the filter to have it not work like a filter and instead work like a search if that is what is desired.

I am averse to having it searchable by the users name atm, because what is stored currently on our end is hkanchi - our db has no idea about what the users name is, there email or anything about them. We also don't really store the usernames as an entity anywhere. 

With the being said it is possible to do.

It is also possible to change the hover to their username as well.

**Comment by ChrisKellaleaMaynard** [comment]
I agree about the filter remaining unchanged, as it is a built-in part of the table and therefore doesn't support paging either.

As confirmed with Clinton, the filter's list of full names is populated by a hardcoded list within the Snomio code - not from a Managed Service feed.

For MVP we can just search by usernames, as the Content Team is already comfortable with using them for other forms of communication.  I will raise a story regarding better user name references, which is likely to require a dynamic user list lookup from Managed Service plus mapped full names from a list within Snomio - which may be related to whitelisting of users independently of the project-level permissions maintained within MS.  When this is implemented we can review whether the search/filter functions can/should use these names instead.
